### ISSUE TITLE
command terminated with exit code 137

### ISSUE THREAD

**Issue by lcmgh** [nextest-rs/nextest#380]
Hi! When executing archived tests (on a Kubernetes POD) via `./cargo-nextest nextest run --archive-file /app/test-archive.tar.zst` I am sometimes facing this issue. I first thought it must be an issue on my end because that specific test depends on some other PODs and I might have an issue in there (but actually I'm waiting till all resources are ready). However in case my test crashes I'd expect nextest to panick with a more verbose error message.

```
Extracting 5 binaries (including 1 non-test binaries) and 1 linked paths to /tmp/nextest-archive-34R3a5
   Extracted 36 files to /tmp/nextest-archive-34R3a5 in 0.23s
    Starting 7 tests across 4 binaries
        PASS [   0.004s]                               my-service::service::test_should_not_insert_lower
        PASS [   0.004s] my-service:bin/my-service::service::test_should_not_insert_lower
        PASS [   0.004s]                               my-service::service::test_should_insert_higher
        PASS [   0.004s]                               my-service settings::test_env
        PASS [   0.004s] my-service:bin/my-service settings::test_env
        PASS [   0.005s] my-service:bin/my-service::service::test_should_insert_higher
command terminated with exit code 137
Error: Process completed with exit code 137
```

**Comment by sunshowers** [comment]
Error 137 is an out of memory error—my guess would be that it's your test that's causing the container to run out of memory.

**Comment by sunshowers** [comment]
The message isn't from nextest... it's from your infrastructure.

**Comment by lcmgh** [comment]
Sorry code have googled that easily. Still thanks for the hint!

**Comment by lcmgh** [comment]
@sunshowers still investigating here on my end. Checked the memory metrics of the POD and overall consumption is very low and far away from its limits. Must be another reason. Strange thing is it happens only sometimes.

**Comment by sunshowers** [comment]
I don't really know what's going on then, unfortunately. You're in the best position to debug this (try looking at strace).

I'd be curious to hear what you find.

**Comment by lcmgh** [comment]
So the error occured only during my cicd flow, not if I executed commands directly. The reason is simple, before executing tests I restarted the test pod deployment, waited until it finished but then sometimes got the id of the old deployment's POD which was still there and on the way to disappear completely but until then open for commands :). Adding `--field-selector=status.phase==Running` parameter helped.

`POD=$(kubectl get pod -l app=myapp --field-selector=status.phase==Running -o jsonpath="{.items[0].metadata.name}" -n mynamespace)`

Still thanks for the support.
